[REDESIGN] What is QML section

mariaschuld opened the pull request:
**Summary:**

Part of the repo redesign. Populates the "What is QML" file of the homepage. 

This page is intended for the interested public, CEOs, journalists and the like, who want to either know about quantum machine learning, or about the scientific foundation of PennyLane. 

It should be easy to understand, and explain PennyLane's approach to quantum machine learning, while motivating that it can be used for any optimization task. 

This page is important as a branding of PL.

**Thanks Josh for the pics!**

![Screenshot from 2020-02-19 10-11-19](https://user-images.githubusercontent.com/23052011/74814564-3acf7000-5300-11ea-81e1-d3a9d1fa1764.png)

antalszava commented on whatisqml.rst:
```diff
@@ -1,7 +1,84 @@
+
+:html:`<br>`
+
+The quantum device is used like a neural network: Data gets "loaded" into the quantum state of the
```
```suggestion
The quantum device is used as a neural network: Data gets "loaded" into the quantum state of the
```

antalszava replied:
(Got from grammar checker, for me it wouldn't make too much of a difference :sweat_smile: )

antalszava commented on whatisqml.rst:
```diff
@@ -1,7 +1,84 @@
+:html:`<br>`
+
+The quantum device is used like a neural network: Data gets "loaded" into the quantum state of the
+hardware and then processed. A measurement reveals the outcome of the machine learning model, for example
```
```suggestion
hardware and then processed. A measurement reveals the outcome of the machine learning model, for example,
```

antalszava commented on whatisqml.rst:
```diff
@@ -1,7 +1,84 @@
+:html:`<br>`
+
+While a lot of research focuses on ideal, universal quantum processing units (QPUs) whose development
+is still a thing of the future [REFs], a large share of quantum machine learning - and the share
```
What would the `[REFs]` parts refer to? Locally I'm getting `Citation <citation> is not referenced.` errors for a couple of things, could this be connected to it?

mariaschuld replied:
I am so sorry. Someone my last push did not go through. That also explains why @josh146 cannot see the figures. Pushed it now with references.

josh146 commented on whatisqml.rst:
```diff
@@ -1,7 +1,84 @@
 =================================
 
+The limits of what machines can learn have always been defined by the computer hardware
+we run our algorithms on - for example, the success of modern-day deep learning  with neural networks is
```
```suggestion
we run our algorithms on --- for example, the success of modern-day deep learning  with neural networks is
```

josh146 commented on whatisqml.rst:
```diff
@@ -1,7 +1,84 @@
+:html:`<br>`
+
+Quantum machine learning extends the pool of hardware for machine learning by an entirely
+new type of computing device - the quantum computer. Information processing with quantum computers
```
```suggestion
new type of computing device --- the quantum computer. Information processing with quantum computers
```

josh146 commented on whatisqml.rst:
```diff
@@ -1,7 +1,84 @@
+
+While a lot of research focuses on ideal, universal quantum processing units (QPUs) whose development
+is still a thing of the future [REFs], a large share of quantum machine learning - and the share
+PennyLane caters for - is interested in near term quantum devices [REFs]. One can understand these devices
```
This sentence seems a bit odd to me, I had to read it a couple of times before it made sense. Maybe worth rewriting?

josh146 commented on whatisqml.rst:
```diff
@@ -1,7 +1,84 @@
+hardware and then processed. A measurement reveals the outcome of the machine learning model, for example
+if an image showed a cat or a dog. The crucial point is that near-term quantum computers are just physical
+machines that are programmed by control parameters, such as an electromagnetic field. By systematically
+adapting these parameters, the quantum computation - and hence the machine learning algorithm - becomes
```
```suggestion
adapting these parameters, the quantum computation --- and hence the machine learning algorithm --- becomes
```

mariaschuld commented:
Sorry @josh146  and @antalszava, apparently the last push did not go through previously. It should be now updated, including your suggestions.

mariaschuld commented:
> Looks good from my end! There is one sentence I noted that could be worth rewriting to make more clear, but the rest reads well. Don't forget to add in the [REFs]!
> 
> I can give the diagrams a go tomorrow in the office laughing

That would be great. The svg versions of the images should be uploaded, not sure if that makes your life easier...

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+
+:html:`<br>`
+
+In the near-term branch of quantum machine learning,
```
Another place where we could say something a bit stronger, e.g., 
```suggestion
In the modern viewpoint on quantum machine learning,
```
could also replace "modern" with "cutting-edge" or "bleeding-edge"

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+**quantum devices are used and trained like neural networks**.
+This is done by systematically adapting the physical control parameters,
+such as an electromagnetic field strength or a laser pulse frequency, to solve a machine learning problem.
+Afterwards, the trained circuit can for example be used to classify the content of images - by encoding
```
```suggestion
For example, the trained circuit can be used to classify the content of images—by encoding
```

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+
+
+But the story does not end here. Quantum circuits turn out to be *differentiable*, which means that a quantum computer
+can compute the change in control parameters needed to become better at a given task. Differentiable programming
```
Can link to wikipedia page for differentiable programming here

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+:html:`<br>`
+
+
+But the story does not end here. Quantum circuits turn out to be *differentiable*, which means that a quantum computer
```
```suggestion
But the story does not end here. Quantum circuits turn out to be *differentiable*, which means that a quantum computer itself
```

co9olguy replied:
Not sure if an improvement or not

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+can compute the change in control parameters needed to become better at a given task. Differentiable programming
+is the very basis of deep learning, implemented in software libraries such as TensorFlow and PyTorch.
+Differentiable programming is also more than deep learning: it is a programming paradigm where steps of an
+algorithm are not hand-coded but learnt.
```
```suggestion
algorithm are not hand-coded but learned.
```
:stuck_out_tongue_winking_eye:

co9olguy replied:
One could argue that the general high-level steps of the algorithm (i.e., the layers, the gates, the ops, etc.) are hand-coded, but left flexible so that they can later be tuned to specific purposes based on data. Not sure how to clearly make that point though, since we also do want to emphasize the "not hand-designed" angle

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+algorithm are not hand-coded but learnt.
+
+Similarly, the idea of trainable quantum computations is larger than quantum machine learning. It includes,
+and in fact originates from, a field called *quantum chemistry* :cite:`peruzzo2014variational`
```
Starting to think we need to shift away from citing references on these types of pages (which are not academic documents) and finding more suitable ways to point to relevant content (e.g., hyperlinks, wikipedia pages, etc)

josh146 replied:
Agree, especially because sphinx-bibtex is not the best plugin. Links that go straight to the source, without an inbetween page of references, is also more user friendly.

mariaschuld replied:
For here I will try to keep paper references to a minimum, i.e. only in the last section. Or shall I replace entirely?

mariaschuld commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+Quantum machine learning is a field at the intersection of artificial intelligence and physics,
+which investigates a new way in which machines can learn from data.
+
+Replacing classical hardware by quantum computers
```
Last one sounds best

mariaschuld commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+:html:`<br>`
+
+In the near-term branch of quantum machine learning,
+**quantum devices are used and trained like neural networks**.
```
Made all three bold...

mariaschuld commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+    :width: 70%
+    :target: javascript:void(0);
+
+    Quantum algorithms can be trained and used like neural networks.
```
No, I swapped them back in the caption

josh146 commented:
<h3>Website build</h3>
<strong>Commit:</strong> 66a6cd17c412d213d12adeca17478e0597b26d2f

<strong>Circle build number:</strong> 448

<strong>Website build:</strong> <a href=https://448-214003948-gh.circle-artifacts.com/0/tmp/qml_html.zip>https://448-214003948-gh.circle-artifacts.com/0/tmp/qml_html.zip</a>

<strong>Website zip:</strong> <a href=https://448-214003948-gh.circle-artifacts.com/0/_build/html/index.html>https://448-214003948-gh.circle-artifacts.com/0/_build/html/index.html</a>

mariaschuld commented:
@co9olguy @josh146 

Implemented your suggestions, and did a big polish again with the wonderful new pictures from Josh. I also added a last section on PennyLane, which is supposed to link people to the rest of the pages.

Regarding the FAQ: the paragraphs are written so that search engines may pick them up already as answers to questions. But I gave up mixing FAQ with the other content and suggest we just add it somewhere hidden, so that it is tailormade for searchability. The FAQ could also frame PennyLane by  questions like "What is PennyLane?" "Why is it different from other quantum computing languages?" "What is differentiable quantum programming?" and so on...

mariaschuld commented:
Latest version's screenshot is updated in the initial post of this PR.

co9olguy commented:
What thumbnail will be used in the initial box that users click on to access this page? (currently it's a clumsy black circle thing, which I assume is a placeholder

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
 
+Quantum machine learning investigates the **consequences of using quantum computers for machine learning**.
+
+For example, quantum machine learning researchers want to find out whether quantum computers can speed up the
```
```suggestion
For example, we might want to find out whether quantum computers can speed up the
```

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+and quantum simulators and hardware to give users the power to train quantum circuits themselves.
+
+To find out more, visit the `PennyLane Documentation <https://pennylane.readthedocs.io/en/stable/>`_, or
+check out the gallery of `hands-on demonstrations <https://pennylane.ai/qml/demos.html>`_.
```
link not working

mariaschuld replied:
Both links work in my local build (?). I replaced the second one with an internal reference ``:doc:hands-on demonstrations <demonstrations>``. 

Please confirm if they are still not working...

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+and in fact originates from, other fields like **quantum chemistry** :cite:`peruzzo2014variational`
+:cite:`mcclean2016theory`, **quantum optimization** :cite:`farhi2014quantum`, and extends to a variety of
+applications such as the **design of quantum algorithms** :cite:`anschuetz2018variational`
+or **quantum error correction** :cite:`johnson2017qvector`.
```
The citation links (e.g., `[R14]`) look really out of place with the rest of the flow of this page :frowning:

mariaschuld replied:
I deleted them for now, and we can link glossary pages in the long run!

co9olguy commented on whatisqml.rst:
```diff
@@ -1,7 +1,105 @@
+Machine learning with near-term quantum devices
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. image:: /_static/whatisqml/quantum_devices_ai.png
```
tiny suggestion, but maybe have just one question mark above FT QPUs. 3 question marks makes it seems like we are really confused, rather than just speculating :laughing:

mariaschuld commented:
> What thumbnail will be used in the initial box that users click on to access this page? (currently it's a clumsy black circle thing, which I assume is a placeholder

I hacked two of our in-house artist's icons together:

![brain_board](https://user-images.githubusercontent.com/23052011/75132040-20b9d700-56de-11ea-979b-222a106c3480.png)

At the moment we do not have a "training circuits" section from which part of the new icon is taken (decision whether this is needed still outstanding), but we can redraw an icon when the need arises.

mariaschuld commented:
@josh146 and @co9olguy, ready for re-review.

josh146 commented on whatisqml.rst:
```diff
@@ -1,7 +1,104 @@
+differentiable quantum computation**. It seamlessly integrates classical machine learning libraries with
+and quantum simulators and hardware to give users the power to train quantum circuits themselves.
+
+To find out more, visit the `PennyLane Documentation <https://pennylane.readthedocs.io/en/stable/>`_, or
```
(Gives us the freedom to more easily change the rtds default build if we need to in the future)
```suggestion
To find out more, visit the `PennyLane Documentation <https://pennylane.readthedocs.io>`_, or
```

co9olguy commented:
Whatever we do with the "Training circuits" card, we should make sure to update the graphic to not be similar image/colour to the QML card

mariaschuld commented:
> Whatever we do with the "Training circuits" card, we should make sure to update the graphic to not be similar image/colour to the QML card

Yes, it's already deleted in a separate PR...
